include/dma.h

```diff
@@ -20,15 +20,16 @@
 #ifndef DOSBOX_DMA_H
 #define DOSBOX_DMA_H
 
+#include <functional>
+
 enum DMAEvent {
 	DMA_REACHED_TC,
 	DMA_MASKED,
 	DMA_UNMASKED,
-//	DMA_TRANSFEREND, this shouldn't really be a ignal
 };
 
 class DmaChannel;
-typedef void (* DMA_CallBack)(DmaChannel * chan,DMAEvent event);
+typedef std::function<void(DmaChannel *chan, DMAEvent event)> DMA_CallBack;
```

Comment:
Nitpick: maybe let's use C++ `using` instead of C `typedef`?

```suggestion
using DMA_CallBack = std::function<void(DmaChannel *chan, DMAEvent event)>;
```

Reply:
Absolutely; and I just read about the benefits of `using` over `typedef` here: https://www.internalpointers.com/post/differences-between-using-and-typedef-modern-c.  Very nice flexibility added when dealing with templates!